server/graphql/v2/interface/Account.js

```diff
@@ -529,12 +537,13 @@ const accountFieldsDefinition = () => ({
     },
   },
   policies: {
-    type: new GraphQLList(Policy),
+    type: new GraphQLNonNull(GraphQLJSON),
```

Comment:
I would advocate for something typed there. JSON is great for flexibility, but then there's no way for frontends to know what to expect form this field. Also, that looks like a breaking change?

Reply:
My idea is that each policy can have its own structure to store its specific parameters.
I think this could definitely be typed, but I don't think that, without a typed frontend, it will make much of a difference since we'll need to reproduce these structures using React's PropTypes.

This is a breaking change that is being addressed in this same PR and includes a migration to update existing data.

Reply:
Also, I first thought it would be an elegant solution to have policies as a polymorphic value that could be either true/false or, in the case it was true, an object that represented the policy parameters:
```js
{
  policies: {
    SIMPLE_POLICY: true,
    MINIMUM_ADMINS: { numberOfAdmins: 3, freezeCollectives: true },
  }
}
```

Reply:
> without a typed frontend, it will make much of a difference since we'll need to reproduce these structures using React's PropTypes.

It does make a difference with ESLint, which detects when we fetch nested fields that are deprecated/removed or have a typo in their names. This is our public API, so I would see this not from the point of view of "What are our frontend needs?" but rather "What would a developer (_internal or external_) expect here?".

> This is a breaking change that is being addressed in this same PR and includes a migration to update existing data.

But what about the frontend, is it not going to be impacted by this?

> Also, I first thought it would be an elegant solution to have policies as a polymorphic value that could be either true/false or, in the case it was true, an object that represented the policy parameters

Typing doesn't prevent us from doing that. In fact, it will help to detect breaking changes if we ever change the type.

If we don't use something typed here, then I don't see the point of using this `policies` field VS retrieving this info directly in `settings.minimumNumberOfAdmins`.

```es6
const HostPolicies = new GraphQLObjectType({
  name: 'HostPolicies',
  description: 'The policies for a given host',
  fields: () => ({
    SIMPLE_POLICY: { type: GraphQLBoolean, description: '...' },
    MINIMUM_ADMINS: {
      type: new GraphQLObjectType({
        name: 'HostPolicyMinimumAdminsConfig',
        fields: () => ({
          numberOfAdmins: { ... },
          freezeCollectives: {}
        })
      })
    }
  })
})
```

Reply:
> But what about the frontend, is it not going to be impacted by this?

Yes, addressed in https://github.com/opencollective/opencollective-frontend/pull/7771

Cool, I agree and I'll type policies.